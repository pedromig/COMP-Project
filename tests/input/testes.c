void f0_A(void);
void f0_B(void,void);

void f1(int a, double b);



int f1(int a, double b);

int f1(int a, double b){

    return a;
}

int f2(int a, double b);
int f2(int a, double b){

    return b;
}

void f3(void){
    //algo
}

void f4(int a, char b, short c, double d);

int main(void){

    void erro;
    int a;
    char b;
    short c;
    double d;

    int a;
    double b;

    if(!a){ }
    if(!b){ }
    if(!c){ }
    if(!d){ }
    if(!e){ }

    if(h){ }

    if(b = b){ }

    if(a){ }
    if(b){ }
    if(c){ }
    if(d){ }

    int coiso = f3();

    f4(a,a,a,a);
    f4(b,b,b,b);
    f4(c,c,c,c);
    f4(d,d,d,d);
}
