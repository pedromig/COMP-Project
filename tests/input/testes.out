Line 2, col 11: Invalid use of void type in declaration
Line 8, col 5: Conflicting types (got int(int,double), expected void(int,double))
Line 10, col 5: Conflicting types (got int(int,double), expected void(int,double))
Line 18, col 12: Conflicting types (got double, expected int)
Line 29, col 10: Invalid use of void type in declaration
Line 35, col 9: Symbol a already defined
Line 36, col 12: Symbol b already defined
Line 41, col 8: Operator ! cannot be applied to type double
Line 42, col 9: Unknown symbol e
Line 42, col 8: Operator ! cannot be applied to type undef
Line 44, col 8: Unknown symbol h
Line 44, col 8: Conflicting types (got undef, expected int)
Line 51, col 8: Conflicting types (got double, expected int)
Line 53, col 9: Conflicting types (got void, expected int)
Line 58, col 8: Conflicting types (got double, expected int)
Line 58, col 10: Conflicting types (got double, expected char)
Line 58, col 12: Conflicting types (got double, expected short)
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
f0_A	void(void)
f1	void(int,double)
f2	int(int,double)
f3	void(void)
f4	void(int,char,short,double)
main	int(void)

===== Function f2 Symbol Table =====
return	int
a	int	param
b	double	param

===== Function f3 Symbol Table =====
return	void

===== Function main Symbol Table =====
return	int
a	int
b	char
c	short
d	double
coiso	int

Program
..FuncDeclaration
....Void
....Id(f0_A)
....ParamList
......ParamDeclaration
........Void
..FuncDeclaration
....Void
....Id(f0_B)
....ParamList
......ParamDeclaration
........Void
......ParamDeclaration
........Void
..FuncDeclaration
....Void
....Id(f1)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Double
........Id(b)
..FuncDeclaration
....Int
....Id(f1)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Double
........Id(b)
..FuncDefinition
....Int
....Id(f1)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Double
........Id(b)
....FuncBody
......Return
........Id(a)
..FuncDeclaration
....Int
....Id(f2)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Double
........Id(b)
..FuncDefinition
....Int
....Id(f2)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Double
........Id(b)
....FuncBody
......Return
........Id(b) - double
..FuncDefinition
....Void
....Id(f3)
....ParamList
......ParamDeclaration
........Void
....FuncBody
..FuncDeclaration
....Void
....Id(f4)
....ParamList
......ParamDeclaration
........Int
........Id(a)
......ParamDeclaration
........Char
........Id(b)
......ParamDeclaration
........Short
........Id(c)
......ParamDeclaration
........Double
........Id(d)
..FuncDefinition
....Int
....Id(main)
....ParamList
......ParamDeclaration
........Void
....FuncBody
......Declaration
........Void
........Id(erro)
......Declaration
........Int
........Id(a)
......Declaration
........Char
........Id(b)
......Declaration
........Short
........Id(c)
......Declaration
........Double
........Id(d)
......Declaration
........Int
........Id(a)
......Declaration
........Double
........Id(b)
......If
........Not - int
..........Id(a) - int
........Null
........Null
......If
........Not - int
..........Id(b) - char
........Null
........Null
......If
........Not - int
..........Id(c) - short
........Null
........Null
......If
........Not - int
..........Id(d) - double
........Null
........Null
......If
........Not - int
..........Id(e) - undef
........Null
........Null
......If
........Id(h) - undef
........Null
........Null
......If
........Store - char
..........Id(b) - char
..........Id(b) - char
........Null
........Null
......If
........Id(a) - int
........Null
........Null
......If
........Id(b) - char
........Null
........Null
......If
........Id(c) - short
........Null
........Null
......If
........Id(d) - double
........Null
........Null
......Declaration
........Int
........Id(coiso)
........Call - void
..........Id(f3) - void(void)
......Call - void
........Id(f4) - void(int,char,short,double)
........Id(a) - int
........Id(a) - int
........Id(a) - int
........Id(a) - int
......Call - void
........Id(f4) - void(int,char,short,double)
........Id(b) - char
........Id(b) - char
........Id(b) - char
........Id(b) - char
......Call - void
........Id(f4) - void(int,char,short,double)
........Id(c) - short
........Id(c) - short
........Id(c) - short
........Id(c) - short
......Call - void
........Id(f4) - void(int,char,short,double)
........Id(d) - double
........Id(d) - double
........Id(d) - double
........Id(d) - double
